src/datasets/arrow_dataset.py

```diff
@@ -1384,6 +1385,44 @@ def with_transform(
         dataset.set_transform(transform=transform, columns=columns, output_all_columns=output_all_columns)
         return dataset
 
+    def prepare_for_task(self, task: Union[str, TaskTemplate]) -> "Dataset":
+        """Prepare a dataset for the given task.
+
+        Casts :attr:`datasets.DatasetInfo.features` according to a task-specific schema.
+
+        Args:
+            task (:obj:`Union[str, TaskTemplate]`): The task to prepare the dataset for during training and evaluation. If :obj:`str`, supported tasks include:
+
+                - :obj:`"text-classification"`
+                - :obj:`"question-answering"`
```

Comment:
`question-answering` exists in two forms: abstractive and extractive question answering.

we can keep a generic `question-answering` but then it will probably mean diferrent schema of input/output for both (abstractive will have text for both while extractive can use spans indication as well as text).

Or we can also propose to use `abstractive-question-answering` and `extractive-question-answering` for instance.

Reply:
Maybe we could have `question-answering-abstractive` and `question-answering-extractive` if somehow we can use a for a completion or search in the future (detail).

Reply:
Actually I see that people are more organizing in terms of general and sub-tasks, for instance on paperwithcode: https://paperswithcode.com/area/natural-language-processing and on nlpprogress: https://github.com/sebastianruder/NLP-progress/blob/master/english/question_answering.md#squad

Probably the best is to align with one of these in terms of denomination, PaperWithCode is probably the most active and maintained and we work with them as well.

Reply:
our idea was to start by following the pipeline taxonomy in `transformers`, where `question-answering` is indeed restricted to the extractive case. 

but you're 100% correct that we should already start considering the abstractive case or grouping in terms of the sub-domains that PwC adopts. 

since our focus right now is on getting `text-classification` integrated and tested with `autonlp`, i've opened an issue  here to tackle this next: https://github.com/huggingface/datasets/issues/2371

Reply:
PWC uses "Question Answering" (meaning extractive) vs "Generative Question Answering" (includes abstractive) which encapsulates most of the QA tasks except Open/Closed Domain QA and Knowledge base QA (they require no context since the knowledge is not part of the query).

I'm fine with these names, or simply extractive and abstractive